platforms/jvm/plugins-java/src/main/java/org/gradle/jvm/component/internal/DefaultJvmSoftwareComponent.java

```diff
@@ -57,26 +57,30 @@ public DefaultJvmSoftwareComponent(
 
     @Override
     public void withJavadocJar() {
-        JvmFeatureInternal feature = getMainFeature();
-
-        feature.withJavadocJar();
-
-        Configuration javadocElements = feature.getJavadocElementsConfiguration();
-        if (!isRegisteredAsLegacyVariant(javadocElements)) {
-            addVariantsFromConfiguration(javadocElements, new JavaConfigurationVariantMapping("runtime", true));
-        }
+        getFeatures().configureEach(feature -> {
```

Comment:
configureEach + checking for the feature name looks very weird here. 

For the sake of having targeted PRs and not changing behavior I think this is fine for now.

Though I think adding a TODO here saying that this _should_ apply to all features would not be a bad idea

Applies below